docs/modules/spam_checker_callbacks.md

```diff
@@ -11,22 +11,29 @@ The available spam checker callbacks are:
 ### `check_event_for_spam`
 
 _First introduced in Synapse v1.37.0_
+_Signature extended to support Allow and Code in Synapse v1.60.0_
+_Boolean return value deprecated in Synapse v1.60.0_
 
 ```python
-async def check_event_for_spam(event: "synapse.events.EventBase") -> Union[bool, str]
+async def check_event_for_spam(event: "synapse.events.EventBase") -> Union[Allow, Code, DEPRECATED_STR, DEPRECATED_BOOL]
```

Comment:
Some comments on this line:

* I'm not convinced the `DEPRECATED_*` types add more clarity than confusion here, especially since we already mentioned above and below that they're deprecated. I'd just replace them with `str` and `bool`.
* For all non-standard types (i.e. types that are custom to Synapse) we want to include the full type to the path, e.g. `"synapse.module_api.errors.Code"`, please do this for `Allow` and `Code`.
* While we're there, can you please update the type of `event` to be `"synapse.module_api.EventBase"`? Since it's now exposed in `synapse.module_api` and Synapse modules should not be importing from outside of this module.